test/extensions/filters/udp/dns_filter/dns_filter_fuzz_test.cc

```diff
@@ -0,0 +1,68 @@
+#include "common/common/logger.h"
+
+#include "extensions/filters/udp/dns_filter/dns_filter.h"
+
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/environment.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace DnsFilter {
+namespace {
+
+const std::string generateQuery(FuzzedDataProvider* data_provider) {
+  size_t query_size = data_provider->ConsumeIntegralInRange<size_t>(0, 512);
+  return data_provider->ConsumeRandomLengthString(query_size);
+}
+
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+  FuzzedDataProvider data_provider(buf, len);
+  const bool recurse = data_provider.ConsumeIntegralInRange<bool>(0, 1);
+  const uint16_t retry_count = data_provider.ConsumeIntegralInRange<uint16_t>(0, 3);
+
+  NiceMock<Runtime::MockRandomGenerator> random;
+  ON_CALL(random, random()).WillByDefault(Return(7));
+  DnsMessageParser message_parser(recurse, retry_count, random);
```

Comment:
Actually, why construct the `MockRandomGenerator` at all? Mocks and setting expectations can slow down a fuzzer, why not consume an integer from the provider for the `random` arg?

Reply:
The `RandomGenerator` object is used in the parser.   I haven't found an example of getting around this.  I see that the `conn_manager_impl_fuzz_test.cc` uses a `NiceMock<Runtime::MockRandomGenerator>` as well.